tests/Doctrine/Tests/ORM/Tools/Pagination/PaginatorTest.php

```diff
@@ -116,4 +120,123 @@ private function createPaginatorWithExtraParametersWithoutOutputWalkers(array $w
 
         return (new Paginator($query, true))->setUseOutputWalkers(false);
     }
+
+    /** @dataProvider dataRedunandQueryPartsAreRemovedForWhereInWalker */
+    public function testRedunandQueryPartsAreRemovedForWhereInWalker(
+        string $dql,
+        array $params,
+        array $expectedQueries
+    ): void {
+        $this->hydrator->method('hydrateAll')->willReturn([[10]]);
+
+        $query = new Query($this->em);
+        $query->setDQL($dql);
+        $query->setParameters($params);
+
+        $query->setMaxResults(1);
+        $paginator = (new Paginator($query, true))->setUseOutputWalkers(false);
+
+        $queryIndex   = 0;
+        $resultMock   = $this->createMock(Result::class);
+        $assertEquals = [$this, 'assertEquals'];
+        $this->connection
+            ->expects($this->exactly(2))
+            ->method('executeQuery')
+            ->willReturnCallback(static function (string $actualSql) use ($expectedQueries, $resultMock, $assertEquals, &$queryIndex): Result {
+                $expectedSql = preg_replace('!\s+!', ' ', $expectedQueries[$queryIndex]);
+                call_user_func($assertEquals, $expectedSql, $actualSql);
```

Comment:
```suggestion
                self::assertEquals($expectedSql, $actualSql);
```

Reply:
Nice catch! Fixed in https://github.com/doctrine/orm/pull/10917/commits/54b3827e955c4f6acbc9ccee39b67a25bec08002.